### DISCUSSION TITLE
iram1_0_seg overflowed when building with frozen library

### DISCUSSION THREAD

**Discussion by wovo** [micropython/micropython#14062] [ESP8266]
I try to include my library as frozen code in a micropyton image. So far this worked OK for rp2040 and esp32. With esp8266 I followed https://github.com/micropython/micropython/tree/master/ports/esp8266 but ran into:

    xtensa-lx106-elf-ld: region `iram1_0_seg' overflowed by 1788 bytes

How can I solve this?

**Comment by robert-hh** [comment]
iram1_0_seg has a size of 32k. It holds static data and time critical code executed from RAM, like Interrupt service. I do not know what's in your library that will be placed there. Maybe viper or native code.

**Reply by wovo** [reply]
Thanks for the tip, I have some native/viper, I'll try without those.

**Reply by wovo** [reply]
Thanks, this step works now! Ran into a next problem, but that seems solvable.
